### PR TITLE
Container import and sync

### PR DESCRIPTION
**Pull request by cjellick**

### CODE REVIEW

File: tests/test_docker.py [hunk 1]
```diff
@@ -822,3 +825,59 @@ def post(req, resp):
...
+        c = get_container('/c861f990-4472-4fa1-960f-65171b544c28')
+        assert c is None
+    event_test(agent, 'docker/instance_remove', post_func=post)
+
+    # Test finding and removing by externalId instead of uuid
+    instance_only_activate(agent, responses)
+    container = get_container('/c861f990-4472-4fa1-960f-65171b544c28')
+    assert container is not None
+
+    def pre(req):
```
Comment (sonchang) [inline tests/test_docker.py]: Does this pre function get used?  we don't specify it in the event_test() below

File: tests/test_docker.py [hunk 2]
```diff
@@ -822,3 +825,59 @@ def post(req, resp):
...
+    # Assert that you can call on a stop container without issue
+    event_test(agent, 'docker/instance_force_stop',
+               pre_func=pre, post_func=post, no_diff=True)
+
+    # And a non-existent one
+    client.remove_container(c)
+    event_test(agent, 'docker/instance_force_stop', pre_func=pre, no_diff=True)
+
+
+def test_instance_remvoe(agent, responses):
```
Comment (sonchang) [inline tests/test_docker.py]: typos with 'remvoe'

File: cattle/plugins/docker/compute.py [hunk 3]
```diff
@@ -92,25 +92,86 @@ def _add_instances(self, ping, pong):
...
-            for name in names:
-                if name.startswith('/'):
-                    name = name[1:]
-                    containers.append({
-                        'type': 'instance',
-                        'uuid': name,
-                        'state': 'running',
-                        'dockerId': c.get('Id')
-                    })
+        all, stopped = self._get_all_containers_by_state()
```
Comment (alena1108) [inline cattle/plugins/docker/compute.py]: should we rename "all" var? As the current name implies that it container all the containers, but it excludes all the containers belonging to "stopped" var

File: tests/docker/instance_force_stop [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    "instanceForceStop": {
+      "id": "9b41268d3c00c1b8d8ed665b1975688a2049687bf566c07acad02bf2409cf032",
+      "kind": "docker"
+    }
+  },
+  "time": 1431116165921,
+  "timeoutMillis": 15000,
+  "transitioningProgress": null,
+  "context": null
+}
```
Comment (alena1108) [inline tests/docker/instance_force_stop]: Playing flake8 here :) new line missing in this file + instance_force_stop_resp, instance_remove

**Comment by sonchang** [comment]
I had a question regarding the last test, but otherwise, LGTM

File: cattle/plugins/docker/compute.py (hunk 3 above)
Comment (cjellick) [inline cattle/plugins/docker/compute.py]: nice catch...went through different versions of this method and the name is from a previous one.

File: tests/test_docker.py (hunk 1 above)
Comment (cjellick) [inline tests/test_docker.py]: good catch. it needs to be used.

**Comment by alena1108** [comment]
LGTM

**Pull request merged**